loomi/regulariza-diagonal-client
# Resident Details

## Resident Details page

PS: please only review after ReportsPage PR being reviewed

PS2: there's a lot of file changes, but most of them are small refactoring, the main changes of the new page are in the following 7 filles:
- src/pages/ResidentDetails (the new page itself)
- src/lib/components/organisms/DetailsResident, DetailsResidence, DetailsDocuments, DetailsSocioeconomics (these 4 files contain all the fields to be reviewed, only the DetailsResident has a different logic a part from the fields)
- src/lib/components/organisms/Modal, ResidentReviewModal (the Modal component only select which modal will be rendered and the ResidentReviewModal component has the logic of the 'Há erros na pesquisa' modal)

---

**What I did:**

- ResidentDetails page shows all the fields that should be filled by the resident
  - The fields are 'pre approved' and the user can decline a field and request the resident to review it (with a date limit and an optional feedback if necessary)
  - The approve button will not make any request if there's even one single field declined
  - The submit button in the modal will not be able to make a request if there's not field declined or if the user do not filled a date limit to the resident answer
 
- Fixed: z-index of modals were explicited so they can overlap everything
 
**How to test:**

1. In your browser go to [stage](http://project.netlify.app/) or [localhost](http://localhost:3000/)
2. Check if the component/view looks just like the [mockup](https://xd.adobe.com/) in different screen sizes (width between 1024px and 1920px)
3. Test the things that you think are worth testing, even the ones that are not in this description

![PR_ResidentDetails](https://user-images.githubusercontent.com/45184938/126231539-7a073d67-9b88-44a0-bd7a-ba4fc7012445.png)
![PR_ResidentDetailsModal](https://user-images.githubusercontent.com/45184938/126231546-9815357a-b698-404b-8df0-10cafa4fcbc3.png)
